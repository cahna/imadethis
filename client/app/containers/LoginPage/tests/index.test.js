/**
 *
 * Tests for LoginPage
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import { LoginPage } from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';
import configureStore from '../../../configureStore';

describe('<LoginPage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const onChangeUsername = jest.fn();
    const onChangePassword = jest.fn();
    const onSubmitForm = jest.fn();
    const clearForm = jest.fn();

    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <LoginPage
            username=""
            password=""
            onChangeUsername={onChangeUsername}
            onChangePassword={onChangePassword}
            onSubmitForm={onSubmitForm}
            clearForm={clearForm}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
    expect(onChangeUsername).not.toHaveBeenCalled();
    expect(onChangePassword).not.toHaveBeenCalled();
    expect(onSubmitForm).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const onChangeUsername = jest.fn();
    const onChangePassword = jest.fn();
    const onSubmitForm = jest.fn();
    const clearForm = jest.fn();

    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <LoginPage
            username="TestUser"
            password="TestPassword"
            onChangeUsername={onChangeUsername}
            onChangePassword={onChangePassword}
            onSubmitForm={onSubmitForm}
            clearForm={clearForm}
          />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
