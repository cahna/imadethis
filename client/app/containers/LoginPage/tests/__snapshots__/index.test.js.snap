// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage /> Should render and match the snapshot 1`] = `
<div>
  <div>
    <section
      class="Section-ojdkzv-0 CenteredSection-sc-1e2by6n-0 jhRZyk"
    >
      <h2
        class="H2-egphza-0 HsCXl"
      >
        <span>
          This is the LoginPage container!
        </span>
      </h2>
      <form
        class="Form-kd3r69-0 kUCzDC"
      >
        <div>
          <label
            for="username"
          >
            <span>
              Username
            </span>
            <input
              class="Input-sc-19txxuq-0 iygDGi"
              id="username"
              placeholder=""
              type="text"
              value="TestUser"
            />
          </label>
        </div>
        <div>
          <label
            for="password"
          >
            <span>
              Password
            </span>
            <input
              class="Input-sc-19txxuq-0 iygDGi"
              id="password"
              placeholder=""
              type="password"
              value="TestPassword"
            />
          </label>
        </div>
        <div
          class="Wrapper-sc-5dsnpv-0 jpYyqh"
        >
          <a
            class="A-t53uvo-0 imPlZY"
          >
            <span>
              Login
            </span>
          </a>
        </div>
      </form>
    </section>
  </div>
</div>
`;
